### PR TITLE
Create empty USER_END trigger handler

### PR DESCRIPTION
**Pull request by thomshutt**
Writing the contents of the USER_END trigger to a database for analytics.

### CODE REVIEW

File: api/http_internal.go [hunk 1]
```diff
@@ -113,6 +116,9 @@ func NewCatalystAPIRouterInternal(cli config.Cli, vodEngine *pipeline.Coordinato
 	// Handler for USER_NEW triggers
 	broker.OnUserNew(accessControlHandlers.HandleUserNew)
 
+	// Handler for USER_END triggers. Only do this if we have a database
```
Comment (iameli) [inline api/http_internal.go:119]: I assume this is implemented in the handler, not really pertinent information
```suggestion
	// Handler for USER_END triggers.
```

Comment (thomshutt) [reply api/http_internal.go:119]: Good catch, it _was_ here and then I realised it made more sense inside

File: handlers/analytics/user_end.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		"stream_id_duration_s",
+		"protocol",
+		"protocol_duration_s",
+		"ip_address",
+		"ip_address_duration_s",
+		"tags"
+		) values($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13)`
+	_, err := a.db.Exec(
+		insertDynStmt,
+		time.Now().Unix(),               // timestamp_ms
```
Comment (iameli) [inline handlers/analytics/user_end.go:54]: Liar!
```suggestion
		time.Now().UnixMilli(),               // timestamp_ms
```

File: handlers/analytics/user_end.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	return AnalyticsHandler{db: db}
+}
+
+func (a *AnalyticsHandler) HandleUserEnd(ctx context.Context, payload *misttriggers.UserEndPayload) error {
+	// If there's nowhere to write to, this handler is a no-op
+	if a.db == nil {
+		return nil
+	}
+
+	streamNames := strings.Split(payload.StreamNames, ",")
```
Comment (iameli) [inline handlers/analytics/user_end.go:29]: Probably this kind of string-parsing stuff should be handled in the trigger-parsing code, so then the `UserEndPayload` can just contain clean slices of data

**Review by iameli — APPROVED** [review]
LGTM, left one comment that we should probably do the comma string parsing prior to the trigger-handling code but this is entirely shippable

File: handlers/analytics/user_end.go (hunk 3 above)
Comment (thomshutt) [reply handlers/analytics/user_end.go:29]: Yeah great call - in the initial implementation I hadn't realised we weren't just passing everything through as strings to the DB. I'll update now.

**Pull request merged**